boto/vendored/regions/regions.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2014 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
I actually originally thought of it just being in ``vendored`` as well, but the additional ``regions`` module is growing on me. I like the fact that if I am trying to update the vendored code related to regions, I know exactly where to look, especially if we have the exceptions and possibly have more modules as revendoring of this logic occurs.

@jamesls Is there any other concerns about the extra layer of nesting other than it looking necessary?